src/sentry/monitors/tasks/detect_broken_monitor_envs.py

```diff
@@ -44,37 +44,49 @@ def detect_broken_monitor_envs():
         # TODO(davidenwang): When we want to email users, remove this filter
         monitorenvbrokendetection__isnull=True,
     )
-    for open_incident in RangeQuerySetWrapper(
-        open_incidents_qs,
-        order_by="starting_timestamp",
-        step=1000,
-    ):
+    org_ids_with_open_incidents = (
+        open_incidents_qs.all().values_list("monitor__organization_id", flat=True).distinct()
+    )
+
+    for org_id in org_ids_with_open_incidents:
         try:
-            organization = Organization.objects.get_from_cache(
-                id=open_incident.monitor.organization_id
-            )
+            organization = Organization.objects.get_from_cache(id=org_id)
             if not features.has(
                 "organizations:crons-broken-monitor-detection", organization=organization
             ):
                 continue
         except Organization.DoesNotExist:
             continue
 
-        # verify that the most recent check-ins have been failing
-        recent_checkins = (
-            MonitorCheckIn.objects.filter(monitor_environment=open_incident.monitor_environment)
-            .order_by("-date_added")
-            .values("status")[:NUM_CONSECUTIVE_BROKEN_CHECKINS]
+        orgs_open_incidents = (
+            open_incidents_qs.all()
+            .select_related("monitor_environment")
```

Comment:
we will need this for grabbing monitor environment information when composing the email context